scripts/liveValidation.js

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License in the project root for license information.
+
+'use strict';
+
+const utils = require('../test/util/utils'),
+    request = require('request-promise-native'),
+    zlib = require('zlib');
+
+const repoUrl = utils.getRepoUrl(),
+    validationService = "https://app.azure-devex-tools.com/api/validations",
+    branch = utils.getSourceBranch(),
+    processingDelay = 20,
+    isRunningInTraviCI = process.env.MODE === 'liveValidation' && process.env.PR_ONLY === 'true',
+    specsPaths = utils.getFilesChangedInPR(),
+    regex = /resource-manager[\\|\/](.*?)[\\|\/].*?[\\|\/](.*?)[\\|\/]/,
+    successThreshold = 90,
+    validationModels = new Map();
+
+let durationInSeconds = parseInt(process.env.LIVE_VALIDATION_DURATION_IN_MINUTES) * 60;
+if (isNaN(durationInSeconds)) {
+    durationInSeconds = 180;
+}
+
+async function runScript() {
+    // See whether script is in Travis CI context
+    console.log(`isRunningInTraviCI: ${isRunningInTraviCI}`);
+    for (const specPath of specsPaths) {
+        let matchResult = specPath.match(regex);
+
+        if (matchResult === null) {
+            continue;
+        }
+
+        let resourceProvider = matchResult[1];
+        let apiVersion = matchResult[2];
+
+        if (!validationModels.has(resourceProvider)) {
+            validationModels.set(resourceProvider, new Set());
+        }
+
+        validationModels.get(resourceProvider).add(apiVersion);
+    }
+
+    if (validationModels.size === 0) {
+        console.log("Change didn't affect any swagger specs. No validation to be done.");
+        return;
+    }
+
+    let resourceProvider = validationModels.keys().next().value;
+    let apiVersion = validationModels.get(resourceProvider).values().next().value;
+
+    console.log(`Changes detected in a swagger spec.`);
+    console.log(`RP is: ${resourceProvider}`);
+    console.log(`ApiVersion is: ${apiVersion}`);
+    console.log(`Source repo is: ${repoUrl}`);
+    console.log(`Branch is: ${branch}`);
+
+    console.log(`Making the request to the validation service...`);
+
+    let response = await request.post(validationService).form({
+        repoUrl: repoUrl,
+        branch: branch,
+        resourceProvider: resourceProvider,
+        apiVersion: apiVersion,
+        duration: durationInSeconds
+    });
+    let validationId = JSON.parse(response).validationId;
+
+    let validationResultUrl = `${validationService}/${validationId}`;
+    console.log(`Request done, results will be available at ${validationResultUrl} in ${durationInSeconds} seconds...`);
```

Comment:
looking at the sample output, not sure that the link in this line is that helpful, it does contain the result, but not easily readable, so you may want to include this link in a "log" line that doesn't lead the user to use it. The way it's included here, seems like the first thing I'd check.

Reply:
That is there just for reference, after the result is returned it displays the results. Take a look at https://travis-ci.org/vladbarosan/azure-rest-api-specs/jobs/325251898 for a sample

Reply:
You might be right to hide it, will update.